### PR TITLE
MLPAB-326 Add validation for donor details and address

### PR DESCRIPTION
**Pull request by hawx**

### CODE REVIEW

File: app/internal/page/your_address_test.go [hunk 1]
```diff
@@ -672,9 +671,38 @@ func TestYourAddressFormValidate(t *testing.T) {
...
-				"address-postcode": "enterPostcode",
+				"address-line-1": "enterAddress",
+				"address-town":   "enterTownOrCity",
+			},
+		},
+		"manual-max-length": {
+			form: &yourAddressForm{
+				Action: "manual",
+				Address: &place.Address{
+					Line1:      strings.Repeat("x", 50),
```
Comment (acsauk) [inline app/internal/page/your_address_test.go:682]: This is handy!

File: app/internal/page/your_details.go [hunk 2]
```diff
@@ -86,21 +96,51 @@ func (d *yourDetailsForm) Validate() map[string]string {
 	if d.FirstNames == "" {
 		errors["first-names"] = "enterFirstNames"
 	}
+	if len(d.FirstNames) > 53 {
+		errors["first-names"] = "firstNamesTooLong"
+	}
```
Comment (acsauk) [inline app/internal/page/your_details.go:101]: Probably not something we can resolve on this PR but do we have the research/reasoning on these specific limits on name lengths? Feels like a hangover from needing to print to a paper form.

**Review by acsauk — APPROVED** [review]
One Q on where the name length limits have come from but we can clear that up as a team if you don't have the info.

File: app/internal/page/your_details.go (hunk 2 above)
Comment (hawx) [reply app/internal/page/your_details.go:101]: I'm just taking them from what was in the spreadsheet shared from Make

**Pull request merged**